File: naivisense/lib/shared/models/therapist.dart
class Therapist {
  final String id;
  final String fullName;
  final String phone;
  final String email;
  final String specialization;
  final int yearsExperience;
  final String city;
  final String? avatarEmoji;
  final String? qualification;
  final String? clinicName;
  final String? clinicAddress;
  final List<String> languages;
  final double rating;
  final int totalSessions;

  const Therapist({
    required this.id,
    required this.fullName,
    required this.phone,
    required this.email,
    required this.specialization,
    required this.yearsExperience,
    required this.city,
    this.avatarEmoji,
    this.qualification,
    this.clinicName,
    this.clinicAddress,
    this.languages = const [],
    this.rating = 0,
    this.totalSessions = 0,
  });

  factory Therapist.fromJson(Map<String, dynamic> json) => Therapist(
        id: json['user_id'] as String? ?? json['id'] as String,
        fullName: json['full_name'] as String? ??
            json['name'] as String? ??
            json['user']?['name'] as String? ??
            json['user']?['full_name'] as String? ??
            '',
        phone:
            json['phone'] as String? ?? json['user']?['phone'] as String? ?? '',
        email:
            json['email'] as String? ?? json['user']?['email'] as String? ?? '',
        specialization: json['specialization'] as String? ?? '',
        yearsExperience: json['years_of_experience'] as int? ??
            json['years_experience'] as int? ??
            0,
        city:
            json['city'] as String? ?? json['clinic_address'] as String? ?? '',
        qualification: json['qualification'] as String?,
        clinicName: json['clinic_name'] as String?,
        clinicAddress: json['clinic_address'] as String?,
        languages: List<String>.from(json['languages'] as List? ?? []),
        rating: (json['rating'] as num?)?.toDouble() ?? 0,
        totalSessions: json['total_sessions'] as int? ?? 0,
      );

  Map<String, dynamic> toJson() => {
        'user_id': id,
        'specialization': specialization,
        'years_of_experience': yearsExperience,
        if (qualification != null) 'qualification': qualification,
        if (clinicName != null) 'clinic_name': clinicName,
        'clinic_address': clinicAddress ?? city,
        'languages': languages,
      };
}

File: backend-node/src/routes/therapist.js
const express = require('express');
const { v4: uuidv4 } = require('uuid');
const db = require('../db');
const { authenticate } = require('../middleware/auth');

const router = express.Router();

router.get('/profile', authenticate, async (req, res) => {
  const { rows } = await db.query(
    'SELECT * FROM therapist_profiles WHERE user_id = $1', [req.user.id]
  );
  if (!rows.length) return res.status(404).json({ detail: 'Therapist profile not found' });
  res.json(rows[0]);
});

router.post('/profile', authenticate, async (req, res) => {
  const existing = await db.query(
    'SELECT id FROM therapist_profiles WHERE user_id = $1', [req.user.id]
  );
  if (existing.rows.length) return res.status(400).json({ detail: 'Profile already exists' });

  const { specialization, years_of_experience, qualification, clinic_name, clinic_address, rating, total_sessions, languages } = req.body;
  const id = uuidv4();
  const { rows } = await db.query(
    `INSERT INTO therapist_profiles
       (id, user_id, specialization, years_of_experience, qualification,
        clinic_name, clinic_address, rating, total_sessions, languages)
     VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10) RETURNING *`,
    [id, req.user.id, specialization, years_of_experience, qualification,
     clinic_name, clinic_address, rating || 0, total_sessions || 0,
     JSON.stringify(languages || [])]
  );
  res.status(201).json(rows[0]);
});

router.put('/profile', authenticate, async (req, res) => {
  const existing = await db.query(
    'SELECT * FROM therapist_profiles WHERE user_id = $1', [req.user.id]
  );
  if (!existing.rows.length) return res.status(404).json({ detail: 'Therapist profile not found' });

  const fields = ['specialization', 'years_of_experience', 'qualification',
                  'clinic_name', 'clinic_address', 'rating', 'total_sessions', 'languages'];
  const updates = [];
  const values = [];
  let idx = 1;

  for (const field of fields) {
    if (req.body[field] !== undefined) {
      updates.push(`${field} = $${idx++}`);
      values.push(field === 'languages' ? JSON.stringify(req.body[field]) : req.body[field]);
    }
  }
  if (!updates.length) return res.json(existing.rows[0]);

  updates.push(`updated_at = NOW()`);
  values.push(req.user.id);
  const { rows } = await db.query(
    `UPDATE therapist_profiles SET ${updates.join(', ')} WHERE user_id = $${idx} RETURNING *`,
    values
  );
  res.json(rows[0]);
});

module.exports = router;